CHANGELOG.md

```diff
@@ -1,3 +1,53 @@
+* **package.json:**  Update dependencies and devDependencies of sdk ([8e07e8f0](https://github.com/Huddly/sdk/commit/8e07e8f0c07df7e7b0b6443b113183f7fa62d44f))
+
+##### Documentation Changes
+
+*  Added documentation for the detector ([#237](https://github.com/Huddly/sdk/pull/237)) ([de0bea1d](https://github.com/Huddly/sdk/commit/de0bea1d3142e9f0befa90162a5690eb6bc3fd54))
+* **Contribution:**  Added description for release process ([7a7399fe](https://github.com/Huddly/sdk/commit/7a7399fef9be93672580a6326b063840a749d4af))
+* **CONTRIBUTING.md:**  Bumped version of compodoc ([e6ca4ffb](https://github.com/Huddly/sdk/commit/e6ca4ffbfdf548330846460917c0131a6033d312))
+
+##### New Features
+
+* **boxfish, huddlygo:**  added current and usb to diagnostics info ([879aefcd](https://github.com/Huddly/sdk/commit/879aefcde3f0abcb9b58cead68d1b1dfba0fe1f4))
+* **Dartfish support:**  Added correct dartfish device class ([02ad6696](https://github.com/Huddly/sdk/commit/02ad6696cc233a64f1218794dcfe99fb8da0c259))
+*  Injectable device factory so you can create custom or extend iDeviceManager clasess ([41176b91](https://github.com/Huddly/sdk/commit/41176b91de803b745bc1162694cf9a8934b97fc1))
+* **getErrorLog:**  Add option for disallowing legacy mode. ([12f730cd](https://github.com/Huddly/sdk/commit/12f730cd153d930f208fba85a48e52a8ed397ceb))
+* **detector:**  Detector should not be dependent on autozoom with newer camera software ([715a5e43](https://github.com/Huddly/sdk/commit/715a5e43176dd61d9a7a9d35649b6157733acd20))
+* **Added options how long to search for device:**  Option for setting # of times looking for device ([056c533f](https://github.com/Huddly/sdk/commit/056c533faac96f8b7c39c7607a0ac99be210a1ae))
+* **Added support for dartfish PID in factory.ts:**  Added support for dartfish PID ([e6c03775](https://github.com/Huddly/sdk/commit/e6c037755cee8d47b5f0f1dbd8e68ca46aa5e856))
+* **Examples:**  Added an example where a specific camera is targeted ([1c88a341](https://github.com/Huddly/sdk/commit/1c88a34137e7b84da248281d8f2c2eb2e363d81f))
+
+##### Bug Fixes
+
+* **api.ts:**  getErrorLog recursive call with exact parameters ([09f6d3ad](https://github.com/Huddly/sdk/commit/09f6d3ad7fd0a359def1b95d079f5fe15b8ea92f))
+* **getErrorLog:**
+  *  Throw exception with error from camera on failure. ([d364036d](https://github.com/Huddly/sdk/commit/d364036d3dd608a58cfe817160037b2155098dd4))
+  *  Don't fall so quickly back to legacy method. Retry once. ([c90c3759](https://github.com/Huddly/sdk/commit/c90c3759dc02c6f606122fcadb59c8a8992c00b7))
+* **examples:**  Update sdk examples to use latest sdk version (0.4.0) ([3b462f73](https://github.com/Huddly/sdk/commit/3b462f73cdd51032b69619c99fc42625c677c8b0))
+
+##### Other Changes
+
+*  add support for PID=0x51 ([5642a2e2](https://github.com/Huddly/sdk/commit/5642a2e2610087cd427cc1f0cbee00278df29cf7))
```

Comment:
Looks like the changelog script isn't working properly! Since 5.0.0 release there are only two new features and no bugfixes, chores or documentation updates

Reply:
What command did you run to generate this changelog?

Reply:
I ran ``` npm run release:minor```

Reply:
But the bump from 0.5.0 to 0.5.1 is not a minor release but a patch release. Regardless, it seems that without specifying which tag to generate the changelog from, it wont work properly. I tried this:
```
./node_modules/generate-changelog/bin/generate --patch -t v0.5.0
```

Reply:
Fixed it I think @brikendr